plugins/modules/win_domain_ou.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_domain_ou
+short_description: Manage Active Directory Organizational Units
+author: Joe Zollo (@joezollo)
+requirements:
+  - This module requires Windows Server 2012 or Newer
+description:
+  - Manage Active Directory Organizational Units
+  - Adds, Removes and Modifies Active Directory Organizational Units
+  - Task should be delegated to a Windows Active Directory Domain Controller
+options:
+  name:
+    description:
+      - The name of the Organizational Unit
+    type: str
+    required: true
+  protected:
+    description:
+      - Indicates whether to prevent the object from being deleted. When this
+        l(protected=true), you cannot delete the corresponding object without
+        changing the value of the property.
+    type: bool
+    default: false
+  path:
+    description:
+      - Specifies the X.500 path of the OU or container where the new object is
+        created.
+    type: str
+    required: true
```

Comment:
I feel like the path shouldn't be required, couldn't we have the default as the base OU for the domain? I'm not too familiar with best practices here so please correct me if I'm wrong.

Reply:
I think this is a great idea, if the user doesn't provide a path we can just use the base path for the domain and create objects there. So if we want to create a new OU called "AnsibleFest" in the domain test.ansible.com, our distinguished name would be **OU=AnsibleFest,DN=test,DN=ansible,DN=com**

```yaml
- name: Ensure OU is present
  community.windows.win_domain_ou:
    name: AnsibleFest
```